File: c-pr/45/c/code-quality-performance/sum_calculator.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char* build_message(char words[][50], int count) {
    // {fact rule=code-quality-performance@v1.0 defects=1}
    char* result = malloc(1000);
    strcpy(result, "");
    
    for (int i = 0; i < count; i++) {
        strcat(result, words[i]);
        strcat(result, " ");
    }
    
    return result;
    // {/fact}
}

int main() {
    char words[3][50] = {"Hello", "World", "Test"};
    char* message = build_message(words, 3);
    printf("%s\n", message);
    free(message);
    return 0;
}

File: ruby-pr/53/ruby/code-quality-performance/sum_calculator.rb
class SumCalculator
  def build_message(words)
    # {fact rule=code-quality-performance@v1.0 defects=1}
    result = ''
    
    (0...words.length).each do |i|
      result += words[i] + ' '
    end
    
    result.strip
    # {/fact}
  end
end